src/init.cpp

```diff
@@ -1544,33 +1556,65 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     ApplyArgsManOptions(args, peerman_opts);
 
     {
-
-        // Read asmap file if configured
-        std::vector<bool> asmap;
+        // Read asmap file or embedded data if configured and initialize
+        // Netgroupman with or without it
+        assert(!node.netgroupman);
+        uint256 asmap_version;
         if (args.IsArgSet("-asmap") && !args.IsArgNegated("-asmap")) {
+            const bool asmap_file_set{args.GetPathArg("-asmap") != ""};
             fs::path asmap_path = args.GetPathArg("-asmap", DEFAULT_ASMAP_FILENAME);
```

Comment:
Wouldn't this be slightly simpler than introducing the `bool asmap_file_set`?
```suggestion
#ifdef ENABLE_EMBEDDED_ASMAP
            fs::path asmap_path = args.GetPathArg("-asmap", "");
#else
            fs::path asmap_path = args.GetPathArg("-asmap", DEFAULT_ASMAP_FILENAME);
#endif
```